### PR TITLE
fix: Fixes from end-to-end tests

### PR DESCRIPTION
**Pull request by aauker**
Updates/changes:
- Regional labeling simplified/performance improvement using GeoCollection
- Tracking 'segment_keys' in the metadata, using them for queries, will be useful when building datasets
- Addressed some edge cases (missing annotations, missing slides, series vs. df with indexing
- Mask sensitive params in logging also
- Allow s3 and s3+https (for regular and secure connections)
- Variable renaming for clarity 
- preprocess.py is now gone!

### CODE REVIEW

File: pyluna-pathology/luna/pathology/cli/generate_tile_labels.py [hunk 1]
```diff
@@ -1,103 +1,129 @@
...
+    df_tiles['intersection_area'] = l_intersection_areas
+
+    logger.info ( df_tiles.loc[df_tiles.intersection_area > 0] )
+
+    output_header_file = f"{output_dir}/{slide_id}.regional_label.tiles.csv"
+    df_tiles.to_csv(output_header_file)        
+    
+    properties = {
+        "slide_tiles": output_header_file, # "Tiles" are the metadata that describe them
+        "segment_keys": {'regional_annotation_id': f"dsa-{collection_name}-{annotation_name}"}
```
Comment (doori) [inline pyluna-pathology/luna/pathology/cli/generate_tile_labels.py:123]: could we use annotation_name or annotation id here? i'd remove the hard-coded "dsa-" here to support annotations generated by other means.

File: pyluna-pathology/luna/pathology/cli/generate_tile_labels.py [hunk 2]
```diff
@@ -1,103 +1,129 @@
...
+            (tile_x+tile_extent,   tile_y+tile_extent),
+            (tile_x+tile_extent,   tile_y),
+        ])
+
+        tile_label = None
+        max_overlap = 0.0
+        for label in d_collections.keys(): 
+            intersection_area = d_collections[label].intersection(tile_polygon).area / tile_polygon.area
+            if intersection_area > max_overlap:
+                tile_label, max_overlap = label, intersection_area
```
Comment (doori) [inline pyluna-pathology/luna/pathology/cli/generate_tile_labels.py:107]: nice!

**Review by doori — APPROVED** [review]
Looks great!

File: pyluna-pathology/luna/pathology/cli/generate_tile_labels.py (hunk 2 above)
Comment (aauker) [reply pyluna-pathology/luna/pathology/cli/generate_tile_labels.py:107]: Thanks :)

File: pyluna-pathology/luna/pathology/cli/dsa_annotation_etl.py [hunk 3]
```diff
@@ -101,7 +101,9 @@ def dsa_annotation_etl(input_dsa_endpoint, username, password, collection_name,
     df_full_annotation_data.to_parquet(slide_annotation_dataset_path)
 
     properties = {
-        "slide_annotation_dataset": slide_annotation_dataset_path
+        "slide_annotation_dataset": slide_annotation_dataset_path,
+        "segment_keys": {"dsa_collection_uuid": collection_uuid}
```
Comment (aauker) [inline pyluna-pathology/luna/pathology/cli/dsa_annotation_etl.py:105]: I add it here now

**Pull request merged**